src/lib/format/ftl/ast/ast.js

```diff
@@ -0,0 +1,120 @@
+class Node {
+  constructor() {
+    this.type = this.constructor.name;
+  }
+}
+
+class Resource extends Node {
+  constructor() {
+    super();
+    this.body = [];
+  }
+}
+
+class Entry extends Node {
+  constructor() {
+    super();
+  }
+}
+
+class Identifier extends Node {
+  constructor(name) {
+    super();
+    this.name = name;
+  }
+}
+
+class String extends Node {
+  constructor(source, elements) {
+    super();
+    this.source = source;
```

Comment:
Could we move `source` from String to Placeable?  I realized that FTL is an opportunity to finally do per-placeable fallback to source, what with all selection logic inside of each placeable.